File: db4ounit/src/db4ounit/tests/RunsLifeCycle.java
package db4ounit.tests;

import db4ounit.*;

public class RunsLifeCycle implements TestCase, TestLifeCycle {

	private boolean _setupCalled=false;
	private boolean _tearDownCalled=false;
	
	public void setUp() {
		_setupCalled=true;
	}
	
	public void tearDown() {
		_tearDownCalled=true;
	}
	
	public boolean setupCalled() {
		return _setupCalled;
	}

	public boolean tearDownCalled() {
		return _tearDownCalled;
	}

	public void testMethod() throws Exception {
		Assert.isTrue(_setupCalled);
		Assert.isTrue(!_tearDownCalled);
		throw FrameworkTestCase.EXCEPTION;
	}
}

File: db4on/Db4oUnit/src/Db4oUnit/tests/RunsLifeCycle.cs
namespace Db4oUnit.tests
{
	public class RunsLifeCycle : Db4oUnit.TestCase, Db4oUnit.TestLifeCycle
	{
		private bool _setupCalled = false;

		private bool _tearDownCalled = false;

		public virtual void SetUp()
		{
			_setupCalled = true;
		}

		public virtual void TearDown()
		{
			_tearDownCalled = true;
		}

		public virtual bool SetupCalled()
		{
			return _setupCalled;
		}

		public virtual bool TearDownCalled()
		{
			return _tearDownCalled;
		}

		public virtual void TestMethod()
		{
			Db4oUnit.Assert.IsTrue(_setupCalled);
			Db4oUnit.Assert.IsTrue(!_tearDownCalled);
			throw Db4oUnit.tests.FrameworkTestCase.EXCEPTION;
		}
	}
}
